### ISSUE TITLE
[Classic: San Andreas] - Events.cleo Plugin Conflict with VehFuncs

### ISSUE THREAD

**Issue by 1234nir** [cleolibrary/CLEO-Redux#87]
[Classic: San Andreas] - Events.cleo Plugin Conflict with [VehFuncs](https://github.com/JuniorDjjr/VehFuncs)

To Reproduce:
1. Install Cleo Redux (Events.cleo must have)
2. Install Vefuncs
3. Others are Optional (CLEO4, SilentPatch and any Others)
4. Start Game > **Game Crash at Haft of Loading Screen**

**If Remove Events.cleo, Game Start Normally**

@x87 @JuniorDjjr

**Comment by JuniorDjjr** [comment]
Send some crash dump (CrashInfo, modloader.log)

**Comment by 1234nir** [comment]
Log Files: [logs.zip](https://github.com/cleolibrary/CLEO-Redux/files/10476676/logs.zip)

>Installed:
Essential Pack (From Mixmods)
CLEO Redux (From Github)
VehFuncs (From Mixmods)

**Comment by JuniorDjjr** [comment]
IPL related... this looks weird.

**Comment by 1234nir** [comment]
@JuniorDjjr main thing

Game works normally with Cleo Redux
**Only Problem with Events.cleo Plugin**

Events.cleo **In**   +   VehFuncs **Out**     = Game Works
VehFuncs **In**  +    Events.cleo **Out**     = Game Works
Events.cleo & VehFuncs (Both In)     = Game Not Works

**Comment by JuniorDjjr** [comment]
And VehFuncs have no relation to events related to IPL, map, unload etc. It's probably some very specific problem, I have nothing in mind.

**Comment by x87** [comment]
Events.cleo hooks two game functions: CWorld::Add and CWorld::Remove. Does VehFuncs call or hook those functions directly or indirectly by spawning something in the game?

**Comment by JuniorDjjr** [comment]
I don't remember why I use 1 byte ahead, but I tried to recompile VehFuncs using VS 2022 and now FxsFuncs (not VehFuncs) crashes on injector inline call for 0x5C235B. I don't know why, even without any changes this happens, I also tried to recompile FxsFuncs in VS 2022.

**Comment by x87** [comment]
I don't see anything useful in 0x5C235B

```
.text:005C2356 1C4 68 00 01 00 00                          push    100h            ; int
.text:005C235B 1C8 8D 94 24 C8 00 00 00                    lea     edx, [esp+1C8h+lineBuf]
.text:005C2362 1C8 52                                      push    edx             ; char *
.text:005C2363 1CC 55                                      push    ebp             ; FILE *
.text:005C2364 1D0 E8 E7 8E EE FF                          call    ReadLine(int,char *,int)
```
maybe you mean 005C2364 ?

**Comment by JuniorDjjr** [comment]
I think the problem existed in my game but I didn't have VehFuncs.asi installed, it doesn't seem to have anything to do recompile, but it's still a very strange problem that I need to check someday (I'm pretty out of modding). Happens with `CFileMgr::ReadLine` inside FxsFuncs in `MakeInline<0x5C235B`.

Here is the VehFuncs recompiled patching `0x00563280` instead, and `push esi`.
https://drive.google.com/file/d/1JzurRk4PWwGpp1Q6gSXpx235n2vDP81y/view?usp=sharing
It works in my game (without using FxsFuncs for some reason), but not tested with CLEO Redux.

**Comment by x87** [comment]
@1234nir can you check that build?

**Comment by 1234nir** [comment]
Hey @x87 & @JuniorDjjr

I just test **VehFuncs Build (Mention in Above)** & Its **Works Fine with Events.cleo** ( I means Complete CLEO Redux)

**Comment by JuniorDjjr** [comment]
So I think it's ok, I'll do another thing and release it for public.

A note about the crash I mentioned in FxsFuncs, for those who are curious, this only happens if I set the `initGameEvent` (0x748CFB) (plugin-sdk) in VehFuncs, then a FxsFuncs patch receives wrong values, it was fixed when changing to another event, I don't understand what could have happened.

**Comment by x87** [comment]
closing as resolved in VehFuncs 2.4
https://www.mixmods.com.br/2022/02/sa-vehfuncs/
